src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Proxy.cs

```diff
@@ -0,0 +1,302 @@
+            AuthenticationSchemes proxyAuthScheme,
+            bool secureServer,
+            bool proxyClosesConnectionAfterFirst407Response)
+        {
+            if (PlatformDetection.IsWindowsNanoServer && IsWinHttpHandler && proxyAuthScheme == AuthenticationSchemes.Digest)
+            {
+                // WinHTTP doesn't support Digest very well and is disabled on Nano.
+                return;
+            }
+
+            if (PlatformDetection.IsFullFramework &&
+                (proxyAuthScheme == AuthenticationSchemes.Negotiate || proxyAuthScheme == AuthenticationSchemes.Ntlm))
+            {
+                // Skip Windows auth tests on .NET Framework.
+                return;
+            }
+
+            if (IsCurlHandler && proxyAuthScheme != AuthenticationSchemes.Basic)
+            {
+                // Issue #27870 curl HttpHandler can only do basic auth to proxy.
+                return;
+            }
+
+            Uri serverUri = secureServer ? Configuration.Http.SecureRemoteEchoServer : Configuration.Http.RemoteEchoServer;
+
+            var options = new LoopbackProxyServer.Options
+                { AuthenticationSchemes = proxyAuthScheme,
+                  ConnectionCloseAfter407 = proxyClosesConnectionAfterFirst407Response
+                };
+            using (LoopbackProxyServer proxyServer = LoopbackProxyServer.Create(options))
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    handler.Proxy = new WebProxy(proxyServer.Uri);
+                    handler.Proxy.Credentials = new NetworkCredential("username", "password");
+                    using (HttpResponseMessage response = await client.GetAsync(serverUri))
+                    {
+                        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                        TestHelper.VerifyResponseBody(
+                            await response.Content.ReadAsStringAsync(),
+                            response.Content.Headers.ContentMD5,
+                            false,
+                            null);                        
+                    }
+                }
+            }
+        }
+
+        [OuterLoop("Uses external server")]
+        [Theory]
+        [MemberData(nameof(CredentialsForProxyRfcCompliant))]
+        public async Task Proxy_BypassFalse_GetRequestGoesThroughCustomProxy_RfcCompliant(ICredentials creds, bool wrapCredsInCache)
+        {
+            await Proxy_BypassFalse_GetRequestGoesThroughCustomProxy_Implementation(creds, wrapCredsInCache);
+        }
+
+        [OuterLoop("Uses external server")]
+        [Theory]
+        [MemberData(nameof(CredentialsForProxyNonRfcCompliant))]
+        public async Task Proxy_BypassFalse_GetRequestGoesThroughCustomProxy_NonRfcCompliant(ICredentials creds, bool wrapCredsInCache)
```

Comment:
Why two identical tests rather than having two MemberDatas on the same test?

Reply:
I moved these tests as-is from the HttpClientHandlerTest.cs file as part of refactoring. But I'll looking into cleanup them up.

Reply:
Thanks; I didn't realize that while reviewing until I got further in the PR.